### PR TITLE
Fix helpdm couldn't DM the user

### LINKED ISSUE #1640: Forbidden: 403 Forbidden (error code: 50007): Cannot send messages to this user
Sentry Issue: [BOT-13C](https://sentry.io/organizations/python-discord/issues/2453738315/?referrer=github_integration)

```
Forbidden: 403 Forbidden (error code: 50007): Cannot send messages to this user
(2 additional frame(s) were not displayed)
...
  File "bot/utils/lock.py", line 108, in wrapper
    log.debug(
  File "bot/utils/lock.py", line 108, in wrapper
    log.debug(
  File "bot/exts/help_channels/_cog.py", line 583, in notify_session_participants
    await message.author.send(embed=embed)
  File "discord/abc.py", line 935, in send
    data = await state.http.send_message(channel.id, content, tts=tts, embed=embed,
  File "discord/http.py", line 241, in request
    raise Forbidden(r, data)

Unhandled exception in on_message.
```

This is caused when a user turns on help dms, but has dms closed.
We should catch this error and deal with it

### PR DESCRIPTION
**Pull request by FaresAhmedb**
I hope that fixes #1640.

### CODE REVIEW

File: bot/exts/help_channels/_cog.py [hunk 1]
```diff
@@ -580,7 +581,24 @@ async def notify_session_participants(self, message: discord.Message) -> None:
...
+                    "Removing user from helpdm."
+                )
+                bot_commands_channel = self.bot.get_channel(Channels.bot_commands)
+                await _caches.help_dm.delete(message.author.id)
+                await bot_commands_channel.send(
+                    f"Hi, {message.author.mention} {constants.Emojis.cross_mark}. "
+                    f"Couldn't DM you helpdm message regarding {message.channel.mention} "
+                    "because your DMs are closed. helpdm has been automatically turned to off. "
+                    "to activate again type `!helpdm on`.",
+                    delete_after=10
```
Comment (Xithrius) [inline bot/exts/help_channels/_cog.py:599]: This integer should be replaced with `RedirectOutput.delete_delay` with `from bot.constants import RedirectOutput`

File: bot/exts/help_channels/_cog.py [hunk 2]
```diff
@@ -580,7 +581,24 @@ async def notify_session_participants(self, message: discord.Message) -> None:
...
+                    f"Failed to send helpdm message to {message.author.id}. DMs Closed/Blocked. "
+                    "Removing user from helpdm."
+                )
+                bot_commands_channel = self.bot.get_channel(Channels.bot_commands)
+                await _caches.help_dm.delete(message.author.id)
+                await bot_commands_channel.send(
+                    f"Hi, {message.author.mention} {constants.Emojis.cross_mark}. "
+                    f"Couldn't DM you helpdm message regarding {message.channel.mention} "
+                    "because your DMs are closed. helpdm has been automatically turned to off. "
+                    "to activate again type `!helpdm on`.",
```
Comment (Xithrius) [inline bot/exts/help_channels/_cog.py:598]: I think this message should be simplified.
```suggestion
                    f"{message.author.mention} {constants.Emojis.cross_mark} "
                    "To receive updates on help channels you're active in, unblock your DMs."
```

**Review by Xithrius — CHANGES_REQUESTED** [review]

File: bot/exts/help_channels/_cog.py [hunk 3]
```diff
@@ -580,7 +581,24 @@ async def notify_session_participants(self, message: discord.Message) -> None:
                 timestamp=message.created_at
             )
             embed.add_field(name="Conversation", value=f"[Jump to message]({message.jump_url})")
-            await message.author.send(embed=embed)
+
+            try:
+                await message.author.send(embed=embed)
+            except discord.errors.ForBidden:
```
Comment (ToxicKidz) [inline bot/exts/help_channels/_cog.py:587]:
```suggestion
            except discord.Forbidden:
```

**Review by ToxicKidz — CHANGES_REQUESTED** [review]

**Comment by Xithrius** [comment]
This PR is in high priority since this feature is getting used relatively often, as far as I can see.

@FaresAhmedb remember to lint before you push by running `poetry run task lint`.

File: bot/exts/help_channels/_cog.py (hunk 2 above)
Comment (dawnofmidnight) [reply bot/exts/help_channels/_cog.py:598]: Unblock may sound like you have the Python bot blocked. "please enable your DMs" or something of the sort may be a bit clearer.

Comment (Xithrius) [reply bot/exts/help_channels/_cog.py:598]: Agreed. I have updated the suggestion.

File: bot/exts/help_channels/_cog.py (hunk 1 above)
Comment (FaresAhmedb) [reply bot/exts/help_channels/_cog.py]: Added it in the last commit https://github.com/python-discord/bot/pull/1642/commits/1b9500b5f7ea0c44ea5023bb264567b05bec1da6. Thanks!

**Comment by FaresAhmedb** [comment]
@Xithrius Sure I will do in the future commits. Thanks for the heads up!

File: bot/exts/help_channels/_cog.py [hunk 4]
```diff
@@ -12,6 +12,8 @@
 
 from bot import constants
 from bot.bot import Bot
+from bot.constants import Channels
+from bot.constants import RedirectOutput
```
Comment (Xithrius) [inline bot/exts/help_channels/_cog.py:16]:
```suggestion
from bot.constants import Channels, RedirectOutput
```

**Comment by Xithrius** [comment]
When committing from a suggestion, please change the message to have a relevant description besides "Updating <file here>".

Context: b9456b4047b658b868c5ea5a965c0610ce7d9736

**Comment by FaresAhmedb** [comment]
@Xithrius Sorry for the mistake. Can I use `--amend`?

**Comment by Xithrius** [comment]
In already public commits, that isn't a good idea. I'll squash merge this PR so there's only one commit in the end.

Thank you for this contribution!

**Review by PureFunctor — APPROVED** [review]

**Review by ToxicKidz — APPROVED** [review]

**Review by Akarys42 — APPROVED** [review]
Looks good to me, thanks a ton for your contribution!

**Pull request merged**